eclipse-che/che-theia
# chore: Adapt Che-Theia to upstream changes (Monaco)

**The PR should be merged at the same time with https://github.com/eclipse/che/pull/20204**

### What does this PR do?
The PR provides an adaptation to the upstream changes related to upgrade the `Monaco` version.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20069

### How to test this PR?

1. `About` dialog should be displayed correctly for both: dark and light themes.
2. Test `Container picker` for `che` tasks:
   -  go to `Terminal` => `Configure Tasks...` and select a `che` task
   -  remove the `component` field from the `che` task configuration
   -  run the task
   -  task system should ask which container should be used for running the task - so it should display list of containers
   -  the task should be run in the selected container 
3. Try to add a plugin registry:
   - go to `F1` => `Plugin Manager: Add Registry`
   - input dropdowns should be displayed to ask a user `Name of your registry` and `Registry URI`  
4. Check binding `QuickAccessContribution` to `ExecTerminalFrontendContribution`:
   - go to `F1` => `View: Show All Opened Terminals`
   - list of opened terminals should be displayed
   - there is known issue:
    https://github.com/eclipse/che/issues/20136
5. Check `TerminalQuickOpenService` related functionality:
   - go to `Terminal` => `Open Terminal in specific container`
   - list of containers should be displayed
   - there are should be `Developer Containers` and `Tooling Containers` categories
   - try to create a terminal for different containers
6. Check the functionality related to Resource monitor:
   - click on the `Workspace resources` item on the bottom panel
   - the list of the corresponding items should be deisplayed
   - a separator is absent due to an issue in upstream:
    https://github.com/eclipse-theia/theia/issues/9758
7. Check the functionality related to `Configure ssh`:
   - add a ssh-reference for a project to your devfile
   - click on `Configure SSH` button when a warning is displayed at cloning a project
   - ssh related items should be displayed, please see a short video
   - a separator is absent due to an issue in upstream:
    https://github.com/eclipse-theia/theia/issues/9758
   
![ssh-items](https://user-images.githubusercontent.com/5676062/127296627-70ac0035-e2d9-4d50-9f93-9641303dea16.gif)
    
8. There are changes for Theme Synchronizer, but it looks like it doesn't work correctly even for the `main` branch of `che-theia`: the system doesn't remember a theme after restarting a workspace/refreshing a page if a user changes it. I think the problem is related to:
 https://github.com/eclipse/che/issues/18287.
 9.  There are changes for `QuickOpenCheWorkspace`, but we have 2 issues for the `main` branch of `che-theia`:
  https://github.com/eclipse/che/issues/19935
  https://github.com/eclipse/che/issues/19936

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
